File: android/src/main/java/com/yolo/utils/FrameValidator.kt
package com.yolo.utils

import android.util.Log
import com.margelo.nitro.camera.HybridFrameSpec


object FrameValidator {
    private const val TAG = "YOLO_TAG_FrameValidator"
    fun isValidYuv(frame: HybridFrameSpec): Boolean {
        if (!frame.isValid) {
            Log.e(TAG, "❌ Frame is not valid")
            return false
        }

        val planes = frame.getPlanes()
        if (planes.size < 3) {
            Log.e(TAG, "❌ Expected 3 YUV planes, got ${planes.size}")
            return false
        }

        planes.forEachIndexed { index, plane ->
            if (!plane.isValid) {
                Log.e(TAG, "❌ Plane $index is not valid")
                return false
            }
        }
        return true
    }
}

File: ios/Utils/FrameValidator.swift
import Foundation
import NitroModules
import VisionCamera

public enum FrameValidator {
    private static let tag = "YOLO_TAG_FrameValidator"
    
    public static func isValidYuv(frame: any HybridFrameSpec) -> Bool {
        // 1. Vérification de la validité globale du Frame
        if !frame.isValid {
            NSLog("[%@]: ❌ Frame is not valid", tag)
            return false
        }
        
        do {
            let planes = try frame.getPlanes()
            
            // 2. Sur iOS, le format matériel natif NV12 contient exactement 2 plans.
            // On s'assure d'avoir au moins ces 2 plans requis.
            if planes.count < 2 {
                NSLog("[%@]: ❌ Expected at least 2 YUV planes (NV12), got %d", tag, planes.count)
                return false
            }
            
            // 3. Parcours et validation de chaque plan individuel
            for (index, plane) in planes.enumerated() {
                if !plane.isValid {
                    NSLog("[%@]: ❌ Plane %d is not valid", tag, index)
                    return false
                }
            }
            
            return true
            
        } catch {
            NSLog("[%@]: ❌ Failed to retrieve planes from frame: %@", tag, error.localizedDescription)
            return false
        }
    }
}
